### PR TITLE
[lesson 4] added small lib to generate a password

### PR DESCRIPTION
**Pull request by wizzardio**
use node cli.js to work with cli tool

### CODE REVIEW

File: lesson4_node_password_generator/lib/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const random = require('../utils/random');
+
+/**
+ * This function represents a common interface that should be used in order to generate random password.
+ * @param {Number} length - total length of password, default - 14, minimal - 8
+ * @param {Number} uppercase - total length of uppercase characters, default - 1
+ * @param {Number} digits - total length of digits, default - 1
+ * @param {Number} special - total length of special symbols, default - 1
+ */
+const index = (uppercase = 1, digits = 1, special = 1, length = 14) => {
```
Comment (webwin0) [inline lesson4_node_password_generator/lib/index.js]: The function name should be more specific.

File: lesson4_node_password_generator/lib/validator/validateType.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * @returns {Object} {error, isValid} - isValid will be true is error`s length is 0
+ */
+const validateType = (length, uppercase, digits, special) => {
+    const re = /^\d+$/;
+    const errors = {}
+    
+    if (re.test(length) !== true) {
+        errors.length = `Length value should be digit, you entered ${typeof length}`;
+    }
+    if (re.test(uppercase) !== true) {
```
Comment (webwin0) [inline lesson4_node_password_generator/lib/validator/validateType.js]: Code duplication, you can create function.

File: lesson4_node_password_generator/utils/random.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    }
+    return text;
+}
+
+const randomStringLowerCase = (length = 11) => randomizer('abcdefghijklmnopqrstuvwxyz', length);
+const randomStringUpperCase = (length = 1) => randomizer('ABCDEFGHIJKLMNOPQRSTUVWXYZ', length);
+const randomStringWithDigits = (length = 1) => randomizer('0123456789', length);
+const randomStringWithSymbols = (length = 1) => randomizer('(.,/-&?$#@!*<>)', length);
+
+module.exports = {
```
Comment (webwin0) [inline lesson4_node_password_generator/utils/random.js]:
```
module.exports = {
    randomStringLowerCase,
    randomStringUpperCase,
    randomStringWithDigits,
    randomStringWithSymbols
};
```

**Comment by wizzardio** [comment]
@webwin0 thanks for code review. 
I applied your comments.
Can you, please, take a look? 

And sorry for the delay.

**Comment by webwin0** [comment]
Good work!

**Comment by wizzardio** [comment]
bagged and tagged. ready for the next.

**Pull request merged**